department/department_app/tests/test_rest.py

```diff
@@ -0,0 +1,171 @@
+        self.data2.update({'name': 'C'})
+        self.data3 = SpecializationListSerializer(self.specialization).data
+        self.data3.update({'name': 'Sen'})
+        self.data4 = EmployeeListSerializer(self.employee).data
+        self.data4.update({'name': 'Ivanov'})
+
+    def test_can_update_department(self):
+        """Test update department"""
+        response = self.client.put(reverse('department-detail', args=[self.department.id]), self.data1)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_can_update_positions(self):
+        """Test update positions"""
+        response = self.client.put(reverse('positions-detail', args=[self.positions.id]), self.data2)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_can_update_specialization(self):
+        """Test update specialization"""
+        response = self.client.put(reverse('specialization-detail', args=[self.specialization.id]), self.data3)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_can_update_employee(self):
+        """Test update employee"""
+        response = self.client.put(reverse('employee-detail', args=[self.employee.id]), self.data4)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+
+class DeleteTests(APITestCase):
+    """Class test delete"""
+    def setUp(self):
+        self.department = Department.objects.create(department='One')
+        self.specialization = Specialization.objects.create(name='Py')
+        self.positions = Positions.objects.create(name='Stud')
+        self.employee = Employee.objects.create(name='Svirydzetski', date_of_birth='1995-05-05',
+                                                department=self.department, positions=self.positions,
+                                                specialization=self.specialization,
+                                                experience=5, wages=20)
+
+    def test_can_delete_department(self):
+        """Test delete department"""
+        response = self.client.delete(reverse('department-detail', args=[self.department.id]))
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
+
+    def test_can_delete_positions(self):
+        """Test delete positions"""
+        response = self.client.delete(reverse('positions-detail', args=[self.positions.id]))
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
+
+    def test_can_delete_specialization(self):
+        """Test delete specialization"""
+        response = self.client.delete(reverse('specialization-detail', args=[self.specialization.id]))
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
+
+    def test_can_delete_employee(self):
+        """Test delete employee"""
+        response = self.client.delete(reverse('employee-detail', args=[self.employee.id]))
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
```

Comment:
1. You should check not only response status code, but make sure that data really removed from the db.
2. Also, should be covered cases when the user is trying to delete record with associated records in the db